tests/core/tests/test_resources.py

```diff
@@ -2570,3 +2570,42 @@ def test_has_natural_foreign_key(self):
                 resources.has_natural_foreign_key(model),
                 expected_result
             )
+
+
+class ManyToManyNullFieldTest(TestCase):
+    # issue 1558 - test cases when m2m fields are null
+    fixtures = ["category", "book"]
+
+    def setUp(self):
+        class _BookM2MResource(resources.ModelResource):
+            class Meta:
+                model = Book
+
+        self.dataset_headers: list[str] = ["id", "categories"]
+        self.book_resource = _BookM2MResource()
+        self.book_resource._meta.skip_unchanged = True
```

Comment:
You could move this assignment to the `Meta` declaration, and I think it would be clearer:

```
class Meta:
    model = Book
    skip_unchanged = True
```